client/components/date-range/test/index.js

```diff
@@ -32,29 +31,24 @@ describe( 'DateRange', () => {
 			};
 		} );
 
-		// Forces the date to be UTC format which avoids offset woes
-		// in the tests
-		fixedEndDate = moment.utc( '2018-06-01' );
-
 		// Set the clock for our test assertions so that new Date()
-		// will return the known `fixedEndDate` set above. This helps
-		// us control the non-determenism that comes from usage of
-		// JS `Date` within the component
-		MockDate.set( fixedEndDate );
+		// will return to a known deterministic date. This important
+		// for the component to render the expected calendars when
+		// an initial date is not passed to it in a test.
+		MockDate.set( moment.utc( '2018-06-01' ) );
```

Comment:
I wonder why we weren't using `jest.setSystemTime()` in the first place - do you have an idea? Perhaps this could be another opportunity to improve the component itself?

Reply:
Yeah, that's a good point. I did try using `jest-setSystemTime()` but it hangs the test and it finally timeouts. Apparently, it's a bug in the version of Jest we currently use: https://github.com/facebook/jest/issues/10221. I tried that when I started working on the tests, so I'll double-check that, just in case.

Reply:
> Perhaps this could be another opportunity to improve the component itself?

If we don't set a fixed date here, some of the tests end up failing with RTL. I haven't dug too much into why, but from what I saw, it's because the rendered calendars will default to the current date if an initial date is not passed. Since we need the actual range we're testing against to actually be rendered and visible in the component for RTL tests, this is important for tests that don't pass the initial date, as they can only use dates in this range (and you'll see that tests that don't set an initial or final selected date will use dates in this range, except perhaps some tests that change the actual date in the inputs).

I think we could avoid this if we always set at least a `firstSelectableDate`, though. I can look into that in a follow-up.

Reply:
I tried replacing the `MockDate.set(...)` with:

```js
jest.useFakeTimers().setSystemTime(moment.utc('06-01-2018', 'MM-DD-YYYY').unix());
```

But the suite hangs and finally fails with a timeout error.

I _think_ we need to [update](https://github.com/facebook/jest/issues/10221#issuecomment-1090157337) to Jest 28. I did try it but got other errors, though :/

Reply:
That being said, I think it's worth exploring the update to Jest 28 in a follow up :)

Reply:
Sounds like a good plan to me, thanks for elaborating 👍